manuscripts/esquery.py

```diff
@@ -431,72 +335,25 @@ def get_agg(cls, field=None, date_field=None, start=None, end=None,
+                agg_id, query_agg = ElasticQuery.__get_query_agg_max(field)
             elif agg_type == "cardinality":
-                query_agg = ElasticQuery.__get_query_agg_cardinality(field)
+                agg_id, query_agg = ElasticQuery.__get_query_agg_cardinality(field)
             elif agg_type == "percentiles":
-                query_agg = ElasticQuery.__get_query_agg_percentiles(field)
+                agg_id, query_agg = ElasticQuery.__get_query_agg_percentiles(field)
             elif agg_type == "avg":
-                query_agg = ElasticQuery.__get_query_agg_avg(field)
+                agg_id, query_agg = ElasticQuery.__get_query_agg_avg(field)
             else:
                 raise RuntimeError("Aggregation of %s not supported" % agg_type)
         else:
-            query_agg = ElasticQuery.__get_query_agg_ts(field, date_field,
-                                                        start=start, end=end,
-                                                        interval=interval,
-                                                        agg_type=agg_type,
-                                                        offset=offset)
-
-        if agg_type not in ['percentiles', 'terms', 'avg']:
-            field_agg = A(agg_type, field=field,
-                          precision_threshold=cls.ES_PRECISION)
-        else:
-            field_agg = A(agg_type, field=field)
-
-        agg_id = cls.AGGREGATION_ID
-
-        if interval:
-            # Two aggs, date histogram and the field+agg_type
-            bounds = ElasticQuery.__get_bounds(start, end)
-            if offset:
-                # With offset and quarter interval bogus buckets are added
-                # to the start and to the end if extended_bounds is used
-                # https://github.com/elastic/elasticsearch/issues/23776
-                bounds = {"offset": offset}
-            ts_agg = A('date_histogram', field=date_field, interval=interval,
-                       time_zone="UTC", min_doc_count=0, **bounds)
-            s.aggs.bucket(agg_id, ts_agg).metric(agg_id + 1, field_agg)
-        else:
-            s.aggs.bucket(agg_id, field_agg)
-
-        query = """
-            {
-              "size": 0,
-              %s,
-              %s
-              }
-        """ % (query_agg, query_basic)
-
-        if USE_ELASTIC_DSL:
-            return json.dumps(s.to_dict())
-        else:
-            return query
+            agg_id, query_agg = ElasticQuery.__get_query_agg_ts(field, date_field,
+                                                                start=start, end=end,
+                                                                interval=interval,
+                                                                agg_type=agg_type,
+                                                                offset=offset)
+        s.aggs.bucket(agg_id, query_agg)
+
+        return json.dumps(s.to_dict())
```

Comment:
I'm not sure why this is returning the query as a JSON file. Shouldn't it better return a es_dsl object, so that we get rid of es DSL queries completely?

Reply:
@jgbarah, that is because in metrics.py, [get_metrics_data](https://github.com/chaoss/grimoirelab-manuscripts/blob/master/manuscripts/metrics/metrics.py#L152) function requests elasticsearch for the data and it requires the data to be in JSON format other wise we will get a 500 error.
![screen shot 2018-05-23 at 2 26 14 pm](https://user-images.githubusercontent.com/9946566/40414435-cb7e3278-5e95-11e8-9e7e-d5501f5929d8.png)


We can change the metrics.py file as well to use the Search().execute method to get the data but i thought that we are probably going to change all the methods and replace them with better more agile methods anyways hence I didn't change all the files.

Reply:
Should I be looking at changing all the files so that we are only using elasticsearch_dsl module?

Reply:
OK, I see. Not for now. Let's keep that for another pr, but let's finish this letting it produce JSON as the result.